--- fltk-1.1.4/src/Makefile.orig	2003-08-24 15:09:06.000000000 +0200
+++ fltk-1.1.4/src/Makefile	2004-04-07 00:16:16.828411328 +0200
@@ -200,7 +200,7 @@
 
 libfltk.so.1.1: $(OBJECTS)
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(OBJECTS)
+	$(DSOCOMMAND) $@ $(OBJECTS) $(LDLIBS)
 	-$(RM) libfltk.so
 	$(LN) libfltk.so.1.1 libfltk.so
 
@@ -236,7 +236,7 @@
 
 libfltk_forms.so.1.1: $(FLOBJECTS) libfltk.so.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(FLOBJECTS) -L. -lfltk
+	$(DSOCOMMAND) $@ $(FLOBJECTS) -L. -lfltk $(LDLIBS) $(IMAGELIBS)
 	-$(RM) libfltk_forms.so
 	$(LN) libfltk_forms.so.1.1 libfltk_forms.so
 
@@ -272,7 +272,7 @@
 
 libfltk_gl.so.1.1: $(GLOBJECTS) libfltk.so.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(GLOBJECTS) -L. -lfltk
+	$(DSOCOMMAND) $@ $(GLOBJECTS) -L. -lfltk $(GLDLIBS)
 	-$(RM) libfltk_gl.so
 	$(LN) libfltk_gl.so.1.1 libfltk_gl.so
 
@@ -308,7 +308,7 @@
 
 libfltk_images.so.1.1: $(IMGOBJECTS) libfltk.so.1.1
 	echo $(DSOCOMMAND) $@ ...
-	$(DSOCOMMAND) $@ $(IMGOBJECTS) $(IMAGELIBS) -L. -lfltk
+	$(DSOCOMMAND) $@ $(IMGOBJECTS) $(IMAGELIBS) -L. -lfltk $(LDLIBS) $(IMAGELIBS)
 	-$(RM) libfltk_images.so
 	$(LN) libfltk_images.so.1.1 libfltk_images.so
 
--- fltk-1.1.4/configure.in.orig	2003-08-06 00:40:45.000000000 +0200
+++ fltk-1.1.4/configure.in	2004-04-07 00:17:07.647685624 +0200
@@ -166,7 +166,7 @@
             FLDSONAME="libfltk_forms.so.$FL_API_VERSION"
             GLDSONAME="libfltk_gl.so.$FL_API_VERSION"
             IMGDSONAME="libfltk_images.so.$FL_API_VERSION"
-            DSOCOMMAND="\$(CXX) -Wl,-soname,\$@ \$(LDLIBS) -shared -fPIC $DEBUGFLAG -o"
+            DSOCOMMAND="\$(CXX) -Wl,-soname,\$@ -shared -fPIC $DEBUGFLAG -o"
 	    if test "x$libdir" != "x/usr/lib"; then
 		    DSOLINK="-Wl,-rpath,$libdir"
             fi
@@ -653,22 +653,7 @@
     # Previous versions of GCC do not have the reliance on the stdc++
     # or g++ libraries, so the extra supc++ library is not needed.
 
-    AC_MSG_CHECKING(if libsupc++ is required)
-
-    SUPC="`$CXX -print-file-name=libsupc++.a 2>/dev/null`"
-    case "$SUPC" in
-    	libsupc++.a*)
-	    # Library not found, so this is and older GCC...
-	    AC_MSG_RESULT(no)
-	    ;;
-	*)
-            # This is gcc 3.x, and it knows of libsupc++, so we need it
-            LIBS="$LIBS -lsupc++"
-            AC_MSG_RESULT(yes)
-	    ;;
-    esac
-
-    CXX="$CC"
+    # PLD: we link all C++ programs against the same version of shared libstdc++.
 
     # Show all standard warnings + unused variables when compiling...
     OPTIM="-Wall -Wunused $OPTIM"
