--- fltk-1.4.3/fluid/Makefile.orig	2025-04-29 13:40:25.000000000 +0200
+++ fltk-1.4.3/fluid/Makefile	2025-06-30 19:06:29.711871474 +0200
@@ -109,8 +109,8 @@ install-linux:
 		fi; \
 		$(INSTALL_DATA) icons/fluid-$$size.png "$(DESTDIR)$(datadir)/icons/hicolor/$${size}x$${size}/apps/fluid.png"; \
 	done
-	-$(INSTALL_DIR) "$(DESTDIR)$(datadir)/mimelnk/application"
-	$(INSTALL_DATA) x-fluid.desktop "$(DESTDIR)$(datadir)/mimelnk/application"
+	-$(INSTALL_DIR) $(DESTDIR)/usr/share/mime/packages
+	$(INSTALL_DATA) fluid.xml $(DESTDIR)/usr/share/mime/packages
 
 install-osx:
 	echo Installing Fluid in $(DESTDIR)/Applications...
--- fltk-1.4.3/fltk-options/Makefile.orig	2025-04-29 13:40:25.000000000 +0200
+++ fltk-1.4.3/fltk-options/Makefile	2025-06-30 19:42:28.936840615 +0200
@@ -67,8 +67,8 @@ install-linux:
 		fi; \
 		$(INSTALL_DATA) icons/fltk-options-$$size.png "$(DESTDIR)/usr/share/icons/hicolor/$${size}x$${size}/apps/fltk-options.png"; \
 	done
-	-$(INSTALL_DIR) "$(DESTDIR)/usr/share/mimelnk/application"
-	$(INSTALL_DATA) x-fltk-options.desktop "$(DESTDIR)/usr/share/mimelnk/application"
+	-$(INSTALL_DIR) "$(DESTDIR)/usr/share/mime/packages"
+	$(INSTALL_DATA) fltk-options.xml "$(DESTDIR)/usr/share/mime/packages"
 
 install-osx:
 	echo Installing fltk-options in $(DESTDIR)/Applications...
