--- fltk-1.3.5/configure.ac.orig	2019-04-10 06:14:17.019965891 +0200
+++ fltk-1.3.5/configure.ac	2019-04-10 06:29:12.975112086 +0200
@@ -1470,7 +1470,7 @@
     if test x$prefix = xNONE; then
 	AC_DEFINE_UNQUOTED(FLTK_DOCDIR, "/usr/local/share/doc/fltk")
     else
-	AC_DEFINE_UNQUOTED(FLTK_DOCDIR, "$prefix/share/doc/fltk")
+	AC_DEFINE_UNQUOTED(FLTK_DOCDIR, "$prefix/share/doc/fltk-apidocs-$FL_VERSION")
     fi
     ;;
 esac
