-- fltk-1.1.8/makeinclude.in~	2008-03-28 22:31:50.280354265 +0100
+++ fltk-1.1.8/makeinclude.in	2008-03-28 22:31:55.920623903 +0100
@@ -127,7 +127,7 @@
 UNINSTALL_DESKTOP = @UNINSTALL_DESKTOP@
 
 # Be quiet when building...
-.SILENT:
+#.SILENT:
 
 # Build commands and filename extensions...
 .SUFFIXES:	.0 .1 .3 .6 .c .cxx .h .fl .man .o .z $(EXEEXT)
--- fltk-1.3.0/configure.in.orig	2011-12-11 18:23:15.000000000 +0100
+++ fltk-1.3.0/configure.in	2011-12-11 18:30:07.000000000 +0100
@@ -335,7 +335,7 @@
 	    ;;
     esac
 
-    LINKSHARED="-L../src $FLTKCAIROOPTION -lfltk_images$SHAREDSUFFIX -lfltk_forms$SHAREDSUFFIX -lfltk$SHAREDSUFFIX"
+    LINKSHARED="-L../src $FLTKCAIROOPTION  -l:../src/libfltk_images.so -l:../src/libfltk_forms.so -l:../src/libfltk.so"
 else
     DSOCOMMAND="echo"
     DSOLINK=""
--- fltk-1.3.0/src/Makefile.orig	2011-12-11 18:23:16.000000000 +0100
+++ fltk-1.3.0/src/Makefile	2011-12-11 18:36:24.000000000 +0100
@@ -557,7 +557,10 @@
 install: $(LIBNAME) $(DSONAME) \
 	$(FLLIBNAME) $(FLDSONAME) \
 	$(GLLIBNAME) $(GLDSONAME) \
-	$(IMGLIBNAME) $(IMGDSONAME) 
+	$(IMGLIBNAME) $(IMGDSONAME)
+	$(RANLIB) $(LIBNAME)
+	$(RANLIB) $(FLLIBNAME)
+	$(RANLIB) $(IMGLIBNAME) 
 	echo "Installing libraries in $(DESTDIR)$(libdir)..."
 	-$(INSTALL_DIR) $(DESTDIR)$(libdir)
 	-$(INSTALL_DIR) $(DESTDIR)$(bindir)
@@ -565,13 +568,10 @@
 	$(INSTALL_LIB) $(LIBNAME) $(DESTDIR)$(libdir)
 	$(INSTALL_LIB) $(FLLIBNAME) $(DESTDIR)$(libdir)
 	$(INSTALL_LIB) $(IMGLIBNAME) $(DESTDIR)$(libdir)
-	$(RANLIB) $(DESTDIR)$(libdir)/$(LIBBASENAME)
-	$(RANLIB) $(DESTDIR)$(libdir)/$(FLLIBBASENAME)
 	if test x$(GLLIBNAME) != x; then \
+		$(RANLIB) $(GLLIBNAME); \
 		$(INSTALL_LIB) $(GLLIBNAME) $(DESTDIR)$(libdir); \
-		$(RANLIB) $(DESTDIR)$(libdir)/$(GLLIBBASENAME); \
 	fi
-	$(RANLIB) $(DESTDIR)$(libdir)/$(IMGLIBBASENAME)
 	if test x$(DSONAME) = xlibfltk.so.1.3; then\
 		$(RM) $(DESTDIR)$(libdir)/libfltk.so*;\
 		$(INSTALL_LIB) libfltk.so.1.3 $(DESTDIR)$(libdir); \
