--- fltk-1.4.3/configure.ac.orig	2025-06-30 19:06:48.688435336 +0200
+++ fltk-1.4.3/configure.ac	2025-06-30 19:09:57.347413283 +0200
@@ -1799,9 +1799,9 @@ AS_CASE([$host_os], [mingw*], [
     AC_DEFINE_UNQUOTED([FLTK_DOCDIR], "$msyspath/local/share/doc/fltk")
 ], [*], [
     AS_IF([test x$prefix = xNONE], [
-        AC_DEFINE_UNQUOTED([FLTK_DOCDIR], "/usr/local/share/doc/fltk")
+        AC_DEFINE_UNQUOTED([FLTK_DOCDIR], "/usr/local/share/doc/fltk-apidocs-$FL_VERSION")
     ], [
-        AC_DEFINE_UNQUOTED([FLTK_DOCDIR], "$prefix/share/doc/fltk")
+        AC_DEFINE_UNQUOTED([FLTK_DOCDIR], "$prefix/share/doc/fltk-apidocs-$FL_VERSION")
     ])
 ])
 
