--- fltk-1.1.7/test/cursor.cxx.orig	2005-04-16 02:13:17.000000000 +0200
+++ fltk-1.1.7/test/cursor.cxx	2006-08-17 09:53:16.305737500 +0200
@@ -39,7 +39,7 @@
 Fl_Hor_Value_Slider *cursor_slider;
 
 void choice_cb(Fl_Widget *, void *v) {
-  cursor = (Fl_Cursor)(int)v;
+  cursor = (Fl_Cursor)(long)v;
   cursor_slider->value(cursor);
   fl_cursor(cursor,fg,bg);
 }
--- fltk-1.1.7/test/input.cxx.orig	2005-11-02 14:57:56.000000000 +0100
+++ fltk-1.1.7/test/input.cxx	2006-08-17 09:55:33.874335000 +0200
@@ -61,7 +61,7 @@
 
 void color_cb(Fl_Widget* button, void* v) {
   Fl_Color c;
-  switch ((int)v) {
+  switch ((long)v) {
   case 0: c = FL_BACKGROUND2_COLOR; break;
   case 1: c = FL_SELECTION_COLOR; break;
   default: c = FL_FOREGROUND_COLOR; break;
--- fltk-1.1.7/test/keyboard.cxx.orig	2005-04-28 19:48:04.000000000 +0200
+++ fltk-1.1.7/test/keyboard.cxx	2006-08-17 09:56:43.650695750 +0200
@@ -105,11 +105,11 @@
     for (int i = 0; i < window->children(); i++) {
       Fl_Widget* b = window->child(i);
       if (b->callback() == (Fl_Callback*)key_cb) {
-	int i = int(b->user_data());
+	int i = long(b->user_data());
 	if (!i) i = b->label()[0];
 	((Fl_Button*)b)->value(Fl::event_key(i));
       } else if (b->callback() == (Fl_Callback*)shift_cb) {
-	int i = int(b->user_data());
+	int i = long(b->user_data());
 	((Fl_Button*)b)->value(Fl::event_state(i));
       }
     }
--- fltk-1.1.7/test/line_style.cxx.orig	2005-04-16 02:13:17.000000000 +0200
+++ fltk-1.1.7/test/line_style.cxx	2006-08-17 09:57:11.348426750 +0200
@@ -54,9 +54,9 @@
   buf[3] = char(sliders[7]->value());
   buf[4] = 0;
   fl_line_style(
-    (int)(choice[0]->mvalue()->user_data()) +
-    (int)(choice[1]->mvalue()->user_data()) +
-    (int)(choice[2]->mvalue()->user_data()),
+    (long)(choice[0]->mvalue()->user_data()) +
+    (long)(choice[1]->mvalue()->user_data()) +
+    (long)(choice[2]->mvalue()->user_data()),
     (int)(sliders[3]->value()),
     buf);
   fl_rect(10,10,w()-20,h()-20);
--- fltk-1.1.7/test/scroll.cxx.orig	2005-04-16 02:13:17.000000000 +0200
+++ fltk-1.1.7/test/scroll.cxx	2006-08-17 09:57:40.554252000 +0200
@@ -71,7 +71,7 @@
 }
 
 void type_cb(Fl_Widget*, void* v) {
-  thescroll->type(int(v));
+  thescroll->type(long(v));
   thescroll->redraw();
 }
 
@@ -87,7 +87,7 @@
 };
 
 void align_cb(Fl_Widget*, void* v) {
-  thescroll->scrollbar.align(int(v));
+  thescroll->scrollbar.align(long(v));
   thescroll->redraw();
 }
 
